core/src/main/java/com/linecorp/armeria/client/HttpSessionHandler.java

```diff
@@ -117,19 +122,44 @@ final class HttpSessionHandler extends ChannelDuplexHandler implements HttpSessi
     @Nullable
     private SessionProtocol retryProtocol;
 
+    /**
+     * {@code true} if an {@link Http2Settings} has been received from the remote endpoint.
+     */
+    private boolean isSettingsFrameReceived;
+    private boolean isChannelActive;
```

Comment:
How about renaming to `channelActivated`, because we don't set it back to false when a channel becomes inactive. We could also add a small comment about this, e.g. `Note: This field never becomes false once it becomes true.`

Reply:
Updated.